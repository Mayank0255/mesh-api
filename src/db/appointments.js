import { performQuery } from ".";

const getAppointmentQuery = `SELECT
	*
FROM
	appointments
WHERE
	id = $1`;

const createAppointmentQuery = `INSERT INTO appointments (type, date, notes, member_id, building_id, request_id, acuity_id)
	VALUES($1, $2, $3, $4, $5, $6, $7)
RETURNING
	*`;

export async function getAppointment(id) {
	return performQuery(getAppointmentQuery, [id]);
}

export async function createAppointment(appointment) {
	return performQuery(createAppointmentQuery, [
		appointment.type,
		appointment.date,
		appointment.notes,
		appointment.member_id,
		appointment.building_id,
		appointment.request_id,
		appointment.acuity_id
	]);
}
