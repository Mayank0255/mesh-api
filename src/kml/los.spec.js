import { performQuery } from "../db";
jest.mock("../db");
import { getLosKML } from "./los";

describe("getLosKML", () => {
  it("renders KML for line-of-sight from a request to other nodes", async () => {
    performQuery.mockResolvedValue([
      {
        requests: [ { id: 5111 } ],
        nodes: [],
        building_a_id: 8331,
        building_b_id: 2034,
        lat_a: '40.645353',
        lng_a: '-74.0152093',
        alt_a: '100m',
        lat_b: '40.6123248',
        lng_b: '-73.9123421',
        alt_b: '150m',
      },
      {
        requests: [ { id: 6111 } ],
        nodes: [ { id: 110 } ],
        building_a_id: 4113,
        building_b_id: 1322,
        lat_a: '39.114023',
        lng_a: '-75.120532',
        alt_a: '50m',
        lat_b: '41.001234',
        lng_b: '-73.144110',
        alt_b: '90m',
      },
    ]);

    expect(await getLosKML()).toMatchSnapshot();
  });
});
