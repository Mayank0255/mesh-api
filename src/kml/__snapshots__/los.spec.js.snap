// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getLosKML renders KML for line-of-sight from a request to other nodes 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<kml xmlns=\\"http://www.opengis.net/kml/2.2\\">
    <Document>
        <Style id=\\"losLink\\">
    <LineStyle>
        <color>7700ff00</color>
        <width>2.5</width>
    </LineStyle>
    <PolyStyle>
        <color>00000000</color>
    </PolyStyle>
</Style>,<Folder><name>5111</name>
<Placemark>
    <name>Line of Sight</name>
    <LineString>
        <altitudeMode>absolute</altitudeMode>
        <coordinates>-74.0152093,40.645353,100m -73.9123421,40.6123248,150m</coordinates>
    </LineString>
    <styleUrl>#losLink</styleUrl>
</Placemark>
	</Folder>,<Folder><name>6111</name>
<Placemark>
    <name>Line of Sight</name>
    <LineString>
        <altitudeMode>absolute</altitudeMode>
        <coordinates>-75.120532,39.114023,50m -73.144110,41.001234,90m</coordinates>
    </LineString>
    <styleUrl>#losLink</styleUrl>
</Placemark>
	</Folder>
    </Document>
</kml>"
`;
